### ISSUE TITLE
Library rendering error react-map-gl

### ISSUE THREAD

**Issue by daviAlves16** [mapbox/mapbox-gl-js#12881]
In react, I defined a default size for the map, but when it renders, the component is the right size, the zoom buttons are all right, but the map image itself starts with a default size of 300 by 400px, and not with what I defined. only after I enlarged the screen a bit, the image goes to the defined size. how to solve this? it does not define the size of the parent div.

```
import Map, {FullscreenControl, GeolocateControl, Marker, Source, Layer, NavigationControl} from "react-map-gl"
import 'mapbox-gl/dist/mapbox-gl.css'

const [viewport, setViewport] = useState({
        latitude: -6.771433,
        longitude: -43.024926,
        zoom: 4,
 });
 <div className='mb-5 mb-xl-8' style={{width: "100%", height: "45rem"}}>
  <Map
        {...viewport}
        ref={mapRef}
        mapLib={import('mapbox-gl')}
        onMove={evt => setViewport(evt.viewState)}
        mapboxAccessToken={TOKEN}
        mapStyle={myStyle}
        style={{width: "100%", height: "45rem"}}
  >    
        <NavigationControl />
        <GeolocateControl />
        <FullscreenControl />    
  </Map>
</div>

**Comment by stepankuzmin** [comment]
Hi @daviAlves16,

Looks like a duplicate of https://github.com/mapbox/mapbox-gl-js/issues/8982

The map's container is most likely resized after the map is rendered. You can resize the map after rendering the component with `map.resize()`. See the original report for more info: https://github.com/mapbox/mapbox-gl-js/issues/8982

I'm closing the issue, but feel free to comment if you have any questions.
